CRM/Admin/Form/MessageTemplates.php

```diff
@@ -214,6 +215,26 @@ public function buildQuickForm() {
     }
   }
 
+  /**
+   * Restrict users access based on permission
+   *
+   * @param int $workflowId
+   */
+  private function checkUserPermission($workflowId) {
+    if (isset($workflowId)) {
```

Comment:
I'm not really familiar with message templates so just a question here, is it impossible for user driven message templates to have a workflow ID?

Reply:
User driven message templates do not have workflow ID. A template becomes system workflow message template the moment it has workflow ID.